src/Helpers/DocumentCreation/DocumentCreationHelper.cs

```diff
@@ -15,18 +16,20 @@ public List<string> GenerateQuestionAndAnswersList(FormAnswers formAnswers, Form
         {
             var summaryBuilder = new SummaryAnswerBuilder();
 
-            var formSchemaQuestions = formSchema.Pages.ToList()
+            var formSchemaQuestions = formSchema.Pages
                 .Where(_ => _.Elements != null)
                 .SelectMany(_ => _.ValidatableElements)
                 .ToList();
 
-            formSchemaQuestions.ForEach(question =>
+            formSchema.Pages.ForEach(page =>
             {
-                var answer = _elementMapper.GetAnswerStringValue(question, formAnswers);
+                formSchemaQuestions.ForEach(question =>
+                {
+                    var answer = _elementMapper.GetAnswerStringValue(question, formAnswers);
+                    summaryBuilder.Add(question.GetLabelText(page.Title), answer, question.Type);
 
-                summaryBuilder.Add(question.GetLabelText(), answer, question.Type);
-
-                summaryBuilder.AddBlankLine();
+                    summaryBuilder.AddBlankLine();
+                });
             });
```

Comment:
not sure if i am readin this wrong, will this not loop round the answers multiple times for each page in the schema, so will add the same question multiple times based on how many pages?

Reply:
You are right! I added the pages loop and missed amending the inner loop